streams/src/test/java/org/apache/kafka/test/MockClientSupplier.java

```diff
@@ -56,8 +57,14 @@ public void setClusterForAdminClient(final Cluster cluster) {
         this.cluster = cluster;
     }
 
+    @SuppressWarnings("deprecation")
     @Override
     public AdminClient getAdminClient(final Map<String, Object> config) {
+        return (AdminClient)getAdmin(config);
```

Comment:
for the mock supplier, we should throw an exception to make sure we don't use it any longer

Reply:
What if some client code uses it?

Reply:
It's test code. Hence, tests should be rewritten accordingly to use the new method IMHO.

Reply:
OK, I can change it to throw ...